napari/_qt/qt_main_window.py

```diff
@@ -1594,14 +1634,57 @@ def _screenshot(
                     add_flash_animation(self._qt_viewer._welcome_widget)
             finally:
                 # make sure we always go back to the right canvas size
-                if size is not None or scale is not None:
+                if size is not None or scale is not None or fit_to_data_extent:
                     canvas.size = prev_size
+                if fit_to_data_extent:
+                    camera.center = old_center
+                    camera.zoom = old_zoom
         else:
             img = self._qt_window.grab().toImage()
             if flash:
                 add_flash_animation(self._qt_window)
         return img
 
+    def export_figure(self, path=None, scale=None, flash=True):
+        """Temporarily reset the view and create a screenshot without margins.
+
+        This function finds a tight boundary around the data, resets the view
+        around that boundary (and, when scale=1, such that 1 captured pixel is
+        equivalent to one data pixel), takes a screenshot, then restores the
+        previous zoom and canvas sizes.
+
+        Parameters
+        ----------
+        path : str
+            Filename for saving screenshot image.
+        scale : float
+            Scale factor used to increase resolution of canvas for the
+            screenshot. By default, the currently displayed resolution. A scale
+            of 1 corresponds to 1 data pixel per screenshot pixel.
+            Only used if `canvas_only` is True.
+        flash : bool
+            Flag to indicate whether flash animation should be shown after
+            the screenshot was captured.
+            By default, True.
+
+        Returns
+        -------
+        image : array
+            Numpy array of type ubyte and shape (h, w, 4). Index [0, 0] is the
+            upper-left corner of the rendered region.
+        """
```

Comment:
I think this needs changing

- A scale of 1 does not mean "1 data pixel per screenshot pixel", but "1 *world* pixel". Relevant especially if layers scales are significantly different from 1.

- "reset view" and "screenshot without margins" is a bit confusing IMO, it's just related to the history of this PR but it's not intuitive out of the box. What margins? I would rather say "export full extent of data as an image".
- "canvas only" reference should be removed

Reply:
Regarding point 1, I understand your point, but it is stated that when scale=1 that is the case. I think instead of talking about world or data we could just give an example.

Reply:
Added an example in the docstring